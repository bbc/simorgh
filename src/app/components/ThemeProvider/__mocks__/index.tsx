import React from 'react';
import { Services, Variants } from '../../../models/types/global';
import defaultServiceVariants from '../defaultServiceVariants';

import afaanoromoo from '../themes/afaanoromoo';
import afrique from '../themes/afrique';
import amharic from '../themes/amharic';
import arabic from '../themes/arabic';
import archive from '../themes/archive';
import azeri from '../themes/azeri';
import bengali from '../themes/bengali';
import burmese from '../themes/burmese';
import cymrufyw from '../themes/cymrufyw';
import gahuza from '../themes/gahuza';
import gujarati from '../themes/gujarati';
import hausa from '../themes/hausa';
import hindi from '../themes/hindi';
import igbo from '../themes/igbo';
import indonesia from '../themes/indonesia';
import japanese from '../themes/japanese';
import korean from '../themes/korean';
import kyrgyz from '../themes/kyrgyz';
import marathi from '../themes/marathi';
import mundo from '../themes/mundo';
import naidheachdan from '../themes/naidheachdan';
import nepali from '../themes/nepali';
import news from '../themes/news';
import newsround from '../themes/newsround';
import pashto from '../themes/pashto';
import persian from '../themes/persian';
import pidgin from '../themes/pidgin';
import portuguese from '../themes/portuguese';
import punjabi from '../themes/punjabi';
import russian from '../themes/russian';
import scotland from '../themes/scotland';
import serbianCyr from '../themes/serbian/cyr';
import serbianLat from '../themes/serbian/lat';
import sinhala from '../themes/sinhala';
import somali from '../themes/somali';
import sport from '../themes/sport';
import swahili from '../themes/swahili';
import tamil from '../themes/tamil';
import telugu from '../themes/telugu';
import thai from '../themes/thai';
import tigrinya from '../themes/tigrinya';
import turkce from '../themes/turkce';
import ukchinaSimp from '../themes/ukchina/simp';
import ukchinaTrad from '../themes/ukchina/trad';
import ukrainian from '../themes/ukrainian';
import urdu from '../themes/urdu';
import uzbek from '../themes/uzbek';
import vietnamese from '../themes/vietnamese';
import yoruba from '../themes/yoruba';
import zhongwenSimp from '../themes/zhongwen/simp';
import zhongwenTrad from '../themes/zhongwen/trad';

type ThemeProvider = {
  [index: string]: React.FC<Props> | { [index: string]: React.FC<Props> };
};

const themeProviders: ThemeProvider = {
  afaanoromoo,
  afrique,
  amharic,
  arabic,
  archive,
  azeri,
  bengali,
  burmese,
  cymrufyw,
  gahuza,
  gujarati,
  hausa,
  hindi,
  igbo,
  indonesia,
  japanese,
  korean,
  kyrgyz,
  marathi,
  mundo,
  naidheachdan,
  nepali,
  news,
  newsround,
  pashto,
  persian,
  pidgin,
  portuguese,
  punjabi,
  russian,
  scotland,
  serbian: {
    cyr: serbianCyr,
    lat: serbianLat,
  },
  sinhala,
  somali,
  sport,
  swahili,
  tamil,
  telugu,
  thai,
  tigrinya,
  turkce,
  ukchina: {
    simp: ukchinaSimp,
    trad: ukchinaTrad,
  },
  ukrainian,
  urdu,
  uzbek,
  vietnamese,
  yoruba,
  zhongwen: {
    simp: zhongwenSimp,
    trad: zhongwenTrad,
  },
};

interface Props {
  children: React.ReactNode;
  service: Services;
  variant: Variants;
}

const ThemeProvider = ({ children, service, ...rest }: Props) => {
  const variant = rest.variant || defaultServiceVariants[service];
  const ThemeProviderSynchronous =
    variant === 'default' || !variant
      ? themeProviders[service]
      : // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore - TODO: come back to this
        themeProviders[service][variant];

  return <ThemeProviderSynchronous>{children}</ThemeProviderSynchronous>;
};

export default ThemeProvider;
