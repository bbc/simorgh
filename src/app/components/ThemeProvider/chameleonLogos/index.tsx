import React from 'react';
import { BrandSVG } from '../../../models/types/theming';

// Chameleon logos have an additional margin on their viewbox
const buildLogo = ({ width, height, group, viewbox }: BrandSVG) => {
  const margin = 140;

  const croppedWidth = (width || 0) - 2 * margin;
  const croppedHeight = (height || 0) - 2 * margin;

  return {
    group: (
      <g
        fillRule="evenodd"
        stroke="#000"
        strokeWidth=".335"
        style={{ stroke: '#fff' }}
      >
        {group}
      </g>
    ),
    ratio: croppedWidth / croppedHeight,
    viewbox: viewbox || {
      minX: margin,
      minY: margin,
      width: croppedWidth,
      height: croppedHeight,
    },
  };
};

export default buildLogo;
