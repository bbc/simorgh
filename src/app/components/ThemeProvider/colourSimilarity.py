colours_compare = [
    '#B80000', 
    '#EB0000', 
    '#FF453A', 
    'FFD230', 
    'FFD53C', 
    '387B12', 
    '#FFFFFF', 
    '#FEFEFE',
    '#F6F6F6',
    '#E6E8EA',
    '#B0B2B4',
    '#8A8C8E',
    '#545658',
    '#3A3C3E',
    '#202224',
    '#141618',
    '#141414',
    '#000000',
    '#0071F1',
    '#3092FF',
    '#0064E6',
    '#009E9E',
    '#00CCC7',
    '#006666',
    '#24B300',
    '#49CC29',
    '#148A00',
    '#FFB32B',
    '#FFCE2B',
    '#C8871E',
    '#F51A5A',
    '#FF4060',
    '#E4134F'
]

def hex_to_rgb(colour):
    colour = colour.lstrip('#')
    return tuple(int(colour[i:i+2], 16) for i in (0, 2, 4))

def euclidian_distance(colour):
    matrix = []
    hex_colour = hex_to_rgb(colour)
    for item in colours_compare:
        compare = hex_to_rgb(item)
        euclidian = 0.3*(compare[0]-hex_colour[0])**2+ 0.59*(compare[1]-hex_colour[1])**2 + 0.11*(compare[2]-hex_colour[2])**2
        matrix.append({"colour": item, "distance": euclidian})
    matrix = sorted(matrix, key=lambda d: d['distance'])
    return matrix

old_colour = "#D5D0CD"
new_colour = euclidian_distance(old_colour)[0]
print("Most similar to " + old_colour + " is "+ new_colour["colour"] + " with dissimilarity of " + str(round(new_colour["distance"])))