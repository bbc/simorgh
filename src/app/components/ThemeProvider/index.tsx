import React, { PropsWithChildren } from 'react';
import { Services, Variants } from '../../models/types/global';
import defaultServiceVariants from './defaultServiceVariants';
import themes from './themes/loadableConfig';

interface Props {
  service: Services;
  variant?: Variants | null;
}

export const ThemeProvider = ({
  children,
  service,
  variant,
}: PropsWithChildren<Props>) => {
  let LoadableContextProvider;

  const serviceVariant: Variants = variant || defaultServiceVariants[service];

  if (serviceVariant === 'default' || !serviceVariant) {
    LoadableContextProvider = themes[service];
  } else {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore - TODO: come back to this
    LoadableContextProvider = themes[service][serviceVariant];
  }

  if (!LoadableContextProvider) {
    return null;
  }

  return <LoadableContextProvider>{children}</LoadableContextProvider>;
};

export default ThemeProvider;
