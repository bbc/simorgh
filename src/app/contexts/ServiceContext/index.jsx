import { createContext } from 'react';
import { node, string } from 'prop-types';
import services from '../../lib/config/services/loadableConfig';
import { getVariant } from '../../lib/utilities/variantHandler';
import getLangOverride from '../../lib/utilities/langHandler';
import variantPropType from '../../models/propTypes/variants';

export const ServiceContext = createContext({});

export const ServiceContextProvider = ({
  children,
  service,
  variant,
  pageLang,
}) => {
  const LoadableContextProvider = services[service];

  if (!LoadableContextProvider) {
    return null;
  }

  return (
    <LoadableContextProvider
      Context={ServiceContext}
      dataKey={
        getLangOverride({ service, pageLang }) ||
        getVariant({ service, variant })
      }
    >
      {children}
    </LoadableContextProvider>
  );
};

ServiceContextProvider.propTypes = {
  children: node.isRequired,
  pageLang: string,
  service: string,
  variant: variantPropType,
};

ServiceContextProvider.defaultProps = {
  pageLang: null,
  service: 'default',
  variant: 'default',
};
