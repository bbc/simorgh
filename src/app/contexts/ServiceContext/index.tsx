import React, { createContext, PropsWithChildren } from 'react';
import services from '../../lib/config/services/loadableConfig';
import { getVariant } from '../../lib/utilities/variantHandler';
import getLangOverride from '../../lib/utilities/langHandler';
import { Services, Variants } from '../../models/types/global';
import { ServiceConfig } from '../../models/types/serviceConfig';

export const ServiceContext = createContext<ServiceConfig>({} as ServiceConfig);

interface Props {
  service: Services;
  variant?: Variants | null;
  pageLang?: string;
}

export const ServiceContextProvider = ({
  children,
  service,
  variant,
  pageLang,
}: PropsWithChildren<Props>) => {
  const LoadableContextProvider = services[service];

  if (!LoadableContextProvider) {
    return null;
  }

  return (
    <LoadableContextProvider
      Context={ServiceContext}
      dataKey={
        getLangOverride({ service, pageLang }) ||
        getVariant({ service, variant })
      }
    >
      {children}
    </LoadableContextProvider>
  );
};
