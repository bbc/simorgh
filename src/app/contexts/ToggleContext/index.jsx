import React, { createContext, useReducer } from 'react';
import { node, object } from 'prop-types';
import defaultToggles from '#lib/config/toggles';
import { toggleReducer } from './reducer';

const environment = process.env.SIMORGH_APP_ENV || 'local';

const ToggleContext = createContext({});

const ToggleContextProvider = ({ children, toggles }) => {
  const [toggleState, toggleDispatch] = useReducer(toggleReducer, toggles);

  return (
    <ToggleContext.Provider value={{ toggleState, toggleDispatch }}>
      {children}
    </ToggleContext.Provider>
  );
};

const ToggleContextConsumer = ToggleContext.Consumer;

ToggleContextProvider.propTypes = {
  children: node.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  toggles: object,
};

ToggleContextProvider.defaultProps = {
  toggles: defaultToggles[environment],
};

export { ToggleContext, ToggleContextProvider, ToggleContextConsumer };
