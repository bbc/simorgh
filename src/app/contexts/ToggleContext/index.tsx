import React, {
  createContext,
  useReducer,
  Dispatch,
  ReducerAction,
  PropsWithChildren,
} from 'react';
import defaultToggles from '#lib/config/toggles';
import { Environments } from '#app/models/types/global';
import toggleReducer from './reducer';

const environment = (process.env.SIMORGH_APP_ENV || 'local') as Environments;

type ToggleContextProps = {
  toggleState: {
    [key: string]: {
      enabled: boolean;
      value?: string;
    };
  };
  toggleDispatch: Dispatch<ReducerAction<typeof toggleReducer>>;
};

const ToggleContext = createContext<ToggleContextProps>(
  {} as ToggleContextProps,
);

type ToggleContextProviderProps = {
  toggles?: {
    [key: symbol]: {
      enabled: boolean;
      value?: string;
    };
  };
};

const ToggleContextProvider = ({
  children,
  toggles = defaultToggles[environment],
}: PropsWithChildren<ToggleContextProviderProps>) => {
  const [toggleState, toggleDispatch] = useReducer(toggleReducer, toggles);

  return (
    <ToggleContext.Provider value={{ toggleState, toggleDispatch }}>
      {children}
    </ToggleContext.Provider>
  );
};

const ToggleContextConsumer = ToggleContext.Consumer;

export { ToggleContext, ToggleContextProvider, ToggleContextConsumer };
