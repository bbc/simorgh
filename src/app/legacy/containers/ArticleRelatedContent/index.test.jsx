import React from 'react';
import { render } from '@testing-library/react';
import { ToggleContextProvider } from '#app/contexts/ToggleContext';
import { ServiceContextProvider } from '../../../contexts/ServiceContext';
import ThemeProvider from '../../../components/ThemeProvider';

import Component, { buildStoryPromos } from '.';

import {
  optimoRelatedContent,
  storyPromoContent,
  relatedContentBlock,
  relatedContentBlockWithTitle,
} from './fixtures';

jest.mock('../../../components/ThemeProvider');

const renderComponent = ({
  content = relatedContentBlock,
  service = 'afrique',
  variant = 'default',
} = {}) =>
  render(
    <ThemeProvider service={service} variant={variant}>
      <ServiceContextProvider service={service} variant={variant}>
        <ToggleContextProvider
          toggles={{
            eventTracking: { enabled: false },
          }}
        >
          <Component content={content} />
        </ToggleContextProvider>
      </ServiceContextProvider>
    </ThemeProvider>,
  );

describe('ArticleRelatedContent', () => {
  it('should render related content', () => {
    const { container, getByText } = renderComponent();

    expect(getByText('News homepage')).toBeInTheDocument();
    expect(
      container.querySelector('a[href="https://www.bbc.co.uk/news"]'),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render custom titles', () => {
    const { getByText } = renderComponent({
      content: relatedContentBlockWithTitle,
    });

    expect(getByText('This is my title')).toBeInTheDocument();
  });

  it('should convert optimo related content blocks to the story promo schema', () => {
    expect(buildStoryPromos(optimoRelatedContent)).toEqual(storyPromoContent);
  });

  it('should add timestamp and update headingTag in the story promo schema', () => {
    const optimoData = optimoRelatedContent;
    const timestamp = 1622753625680;

    optimoData[0].model.blocks.push({
      type: 'aresLink',
      model: {
        blocks: [
          {
            type: 'optimoLinkMetadata',
            model: {
              timestamp,
            },
          },
        ],
      },
    });

    const result = buildStoryPromos(optimoData);

    expect(result[0].headingTag).toEqual('h3');
    expect(result[0].timestamp).toEqual(timestamp);
  });
});
