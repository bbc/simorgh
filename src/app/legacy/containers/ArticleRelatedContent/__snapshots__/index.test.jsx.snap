// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleRelatedContent should render related content 1`] = `
.emotion-1 {
  margin-bottom: 1rem;
}

@media (max-width: 24.9375rem) {
  .emotion-1 {
    padding: 0 0.5rem;
  }
}

@media (min-width: 25rem) and (max-width: 62.9375rem) {
  .emotion-1 {
    padding: 0 1rem;
  }
}

.emotion-4 {
  position: relative;
  z-index: 0;
  color: #141414;
  margin-top: 2rem;
  margin-top: 0;
}

@media (min-width: 37.5rem) {
  .emotion-4 {
    margin-top: 1.5rem;
  }
}

@media (min-width: 63rem) {
  .emotion-4 {
    margin-bottom: 1.5rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-4 {
    margin-top: 0;
  }
}

.emotion-6 {
  margin: 0;
  padding: 0;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-height: 2.75rem;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

@media (min-width: 37.5rem) {
  .emotion-10 {
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

.emotion-12 {
  font-size: 1.125rem;
  line-height: 1.5rem;
  font-family: Helmet,Freesans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  background-color: #F6F6F6;
  margin: 1rem 0;
  padding-right: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-12 {
    font-size: 1.25rem;
    line-height: 1.625rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-12 {
    font-size: 1.5rem;
    line-height: 2rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-12 {
    margin: 0;
  }
}

@media (min-width: 37.5rem) {
  .emotion-12 {
    padding-right: 1rem;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-14 {
    padding-top: 1rem;
  }
}

@media (min-width: 63rem) {
  .emotion-14 {
    padding-bottom: 1.5rem;
  }
}

@supports (display: grid) {
  .emotion-16 {
    display: grid;
    position: initial;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-16 {
      grid-template-columns: repeat(1, 1fr);
      grid-column-end: span 1;
      grid-column-gap: 0.5rem;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-16 {
      grid-template-columns: repeat(1, 1fr);
      grid-column-end: span 1;
      grid-column-gap: 0.5rem;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-16 {
      grid-template-columns: repeat(1, 1fr);
      grid-column-end: span 1;
      grid-column-gap: 0.5rem;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-16 {
      grid-template-columns: repeat(1, 1fr);
      grid-column-end: span 1;
      grid-column-gap: 1rem;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-16 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
      grid-column-gap: 1rem;
    }
  }

  @media (min-width: 80rem) {
    .emotion-16 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
      grid-column-gap: 1rem;
    }
  }
}

.emotion-18 {
  position: relative;
}

@supports (grid-template-columns: fit-content(200px)) {
  .emotion-18 {
    display: grid;
    grid-template-columns: repeat(6, 1fr);
    grid-column-gap: 0.5rem;
  }

  @media (min-width: 37.5rem) {
    .emotion-18 {
      grid-column-gap: 1rem;
    }
  }

  @media (min-width: 63rem) {
    .emotion-18 {
      display: block;
    }
  }
}

.emotion-20 {
  display: inline-block;
  vertical-align: top;
  position: relative;
  width: 33.33%;
}

@media (min-width: 63rem) {
  .emotion-20 {
    display: block;
    width: 100%;
  }
}

@supports (grid-template-columns: fit-content(200px)) {
  .emotion-20 {
    width: initial;
    grid-column: 1/span 2;
  }
}

.emotion-22 {
  position: relative;
}

.emotion-24 {
  height: 0;
  overflow: hidden;
  position: relative;
  background-image: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHdpZHRoPSI0NzkiIGhlaWdodD0iMTM2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPjxnIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+PGcgZmlsbD0iI0Q1RDBDRCI+PGc+PHBhdGggZD0iTTQ0OS41NTYgOTEuNTUzbC0yLjMxIDEuNDFjLTExLjE2NyA2LjgxOC0yMy4zMTMgMTAuNDc0LTM0LjM0NiAxMC40Ny0yMy42MS0uMDktMzkuMTYzLTE0LjA4My0zOS4yMjctMzQuNDUzLjAzLTE5LjkxOCAxNi4yNy0zNC42NjMgMzguNjMzLTM0LjcxOCAxMS4zODcuMDEgMjEuNzAzIDIuOTU0IDMzLjk2MiA5LjY3MmwyLjI1MSAxLjI0di0xOC4xOWwtLjk2Mi0uMzc3Yy0xMy44MjQtNS40NTgtMjQuNTMtNy44OS0zNS4xMDMtNy44ODQtMTYuMzQ2LS4wMDYtMzAuNTMzIDUuMzk0LTQwLjYzNyAxNC41NTctMTAuMTA1IDkuMTYzLTE2LjEwNiAyMi4xMDItMTYuMDk5IDM2Ljk1My4wMDggMTAuMzQ4IDQuMjc5IDIyLjQ4IDEzLjQyIDMyLjEwNSA5LjEyMSA5LjYyOCAyMy4xNjUgMTYuNjQ4IDQyLjQzIDE2LjYzOWguMDYzYzE1Ljk4IDAgMjcuMDYyLTMuNTYzIDM3LjA3NC04LjQ5MmwuODUxLS40MTRWOTEuNTUzek0zMzQgMTM1LjY5N2gxNDQuMTk1VjBIMzM0djEzNS42OTd6Ii8+PHBhdGggZD0iTTI3Ni45MzcgODkuOTY4Yy4wNDEtMTIuMzMtOC4xNzEtMjEuNjk2LTIxLjMwOC0yNS4zIDMuNTQ0LTEuODA5IDYuMzUtNC4wMjMgOC40MDQtNi43MjcgMi43NS0zLjYyMiA0LjA2MS04LjA2NCA0LjA0Ni0xMy4yMzUuMDE1LTYuMzU5LTIuNDg2LTEyLjgzOS03Ljg1OC0xNy42ODctNS4zNzItNC44NDctMTMuNTI2LTcuOTk3LTI0LjY1NC03Ljk5MUgyMDQuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMTY3IDEzNS42OThoMTQ0LjE5N1YwSDE2N3YxMzUuNjk3eiIvPjxwYXRoIGQ9Ik0xMDkuOTM3IDg5Ljk2OGMuMDQxLTEyLjMzLTguMTcxLTIxLjY5Ni0yMS4zMDgtMjUuMyAzLjU0NC0xLjgwOSA2LjM1LTQuMDIzIDguNDA0LTYuNzI3IDIuNzUtMy42MjIgNC4wNjEtOC4wNjQgNC4wNDYtMTMuMjM1LjAxNS02LjM1OS0yLjQ4Ni0xMi44MzktNy44NTgtMTcuNjg3LTUuMzcyLTQuODQ3LTEzLjUyNi03Ljk5Ny0yNC42NTQtNy45OTFIMzcuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMCAxMzUuNjk4aDE0NC4xOTdWMEgwdjEzNS42OTd6Ii8+PHBhdGggZD0iTTI1OC42NjIgODguMTk4Yy0uMDEzIDMuMjI5LTEuMDA3IDYuNDc1LTMuODk2IDkuMDExLTIuODg0IDIuNTM3LTcuODczIDQuNDYzLTE2LjEzMyA0LjQ2M0gyMjJWNzVoMTUuODkzYzcuNDExIDAgMTIuNjcgMS41MDIgMTUuOTY1IDMuODUgMy4yODkgMi4zNjIgNC43NzYgNS40NjMgNC44MDQgOS4zNDgiLz48cGF0aCBkPSJNOTEuNjYyIDg4LjE5OGMtLjAxMyAzLjIyOS0xLjAwNyA2LjQ3NS0zLjg5NiA5LjAxMS0yLjg4NCAyLjUzNy03Ljg3NCA0LjQ2My0xNi4xMzMgNC40NjNINTVWNzVoMTUuODkyYzcuNDEyIDAgMTIuNjcyIDEuNTAyIDE1Ljk2NiAzLjg1IDMuMjg5IDIuMzYyIDQuNzc2IDUuNDYzIDQuODA0IDkuMzQ4Ii8+PHBhdGggZD0iTTI0NS4xODYgNTUuNzljMy4wOTYtMi4yMzcgNC41OS01LjM4NiA0LjYxMy0xMC4xMjQtLjAxNS0zLjI1LS45NDMtNi4wMzMtMy4yODEtOC4xMTEtMi4zNDYtMi4wNzgtNi4zMy0zLjU1NS0xMi43NTQtMy41NTVIMjIydjI1LjI3NWg4LjA3NmM2Ljk4OC4wMDQgMTEuOTk4LTEuMjQzIDE1LjExLTMuNDg2Ii8+PHBhdGggZD0iTTc4LjE4NiA1NS43OWMzLjA5Ni0yLjIzNyA0LjU5LTUuMzg2IDQuNjEzLTEwLjEyNC0uMDE1LTMuMjUtLjk0My02LjAzMy0zLjI4Mi04LjExMUM3Ny4xNzIgMzUuNDc3IDczLjE4OCAzNCA2Ni43NjQgMzRINTV2MjUuMjc1aDguMDc2YzYuOTg4LjAwNCAxMS45OTgtMS4yNDMgMTUuMTEtMy40ODYiLz48L2c+PC9nPjwvZz48L3N2Zz4K);
  background-color: #F2F2F2;
  -webkit-background-position: center center;
  background-position: center center;
  background-repeat: no-repeat;
  -webkit-background-size: 60px 17px;
  background-size: 60px 17px;
}

@media (min-width: 25rem) {
  .emotion-24 {
    -webkit-background-size: 77px 22px;
    background-size: 77px 22px;
  }
}

@media (min-width: 63rem) {
  .emotion-24 {
    -webkit-background-size: 93px 27px;
    background-size: 93px 27px;
  }
}

.emotion-25 {
  width: 100%;
  height: auto;
  object-fit: cover;
}

@media (min-width: 25rem) {
  .emotion-26 {
    position: absolute;
    bottom: 0;
  }
}

.emotion-26>* {
  height: 2rem;
  padding: 0.5rem 0.25rem;
}

@media (max-width: 24.9375rem) {
  .emotion-26>* {
    height: 1.25rem;
    padding: 0.25rem 0.25rem 0;
  }
}

.emotion-28 {
  display: inline-block;
  vertical-align: top;
  width: 66.67%;
  padding: 0 0.5rem;
}

@media (min-width: 37.5rem) {
  .emotion-28 {
    padding: 0 1rem;
  }
}

@media (min-width: 63rem) {
  .emotion-28 {
    display: block;
    width: 100%;
    padding: 0.5rem 0;
  }
}

@supports (grid-template-columns: fit-content(200px)) {
  .emotion-28 {
    display: block;
    width: initial;
    padding: initial;
    grid-column: 3/span 4;
  }

  @media (min-width: 63rem) {
    .emotion-28 {
      padding-top: 0.5rem;
    }
  }
}

.emotion-30 {
  color: #222222;
  margin: 0;
  padding-bottom: 0.5rem;
  font-family: Helmet,Freesans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-30 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-30 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

.emotion-32 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow-wrap: break-word;
  overflow-wrap: anywhere;
}

.emotion-32:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.emotion-32:hover,
.emotion-32:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-32:visited {
  color: #6E6E73;
}

<div>
  <section
    aria-labelledby="related-content-heading"
    class=" emotion-0 emotion-1 emotion-2"
    data-e2e="related-content-heading"
    role="region"
  >
    <div
      class="emotion-3 emotion-4 emotion-5"
    >
      <h2
        class="emotion-6 emotion-7"
      >
        <span
          class="emotion-8 emotion-9"
        >
          <span
            class="emotion-10 emotion-11"
          >
            <span
              class="emotion-12 emotion-13"
              dir="ltr"
              id="related-content-heading"
            >
              Lire plus
            </span>
          </span>
        </span>
      </h2>
    </div>
    <div
      class="emotion-14 emotion-15"
    >
      <div
        class="emotion-16 emotion-17"
        dir="ltr"
      >
        <div
          class="emotion-18 emotion-19"
          data-e2e="story-promo"
        >
          <div
            class="emotion-20 emotion-21"
            dir="ltr"
          >
            <div
              class="emotion-22 emotion-23"
            >
              <div
                class="emotion-24"
                style="padding-bottom: 56.26%;"
              >
                <picture>
                  <img
                    alt="BBC Test Image"
                    class="emotion-25"
                    height="562"
                    loading="lazy"
                    sizes="(min-width: 1020px) 232px, calc(31.86vw - 7px)"
                    src="https://ichef.bbci.co.uk/news/660/cpsdevpb/72c5/test/47abb9d0-5431-11ec-b75b-9f9fe566903d.png"
                    srcset="https://ichef.bbci.co.uk/news/70/cpsdevpb/72c5/test/47abb9d0-5431-11ec-b75b-9f9fe566903d.png 70w, https://ichef.bbci.co.uk/news/95/cpsdevpb/72c5/test/47abb9d0-5431-11ec-b75b-9f9fe566903d.png 95w, https://ichef.bbci.co.uk/news/144/cpsdevpb/72c5/test/47abb9d0-5431-11ec-b75b-9f9fe566903d.png 144w, https://ichef.bbci.co.uk/news/183/cpsdevpb/72c5/test/47abb9d0-5431-11ec-b75b-9f9fe566903d.png 183w, https://ichef.bbci.co.uk/news/240/cpsdevpb/72c5/test/47abb9d0-5431-11ec-b75b-9f9fe566903d.png 240w, https://ichef.bbci.co.uk/news/320/cpsdevpb/72c5/test/47abb9d0-5431-11ec-b75b-9f9fe566903d.png 320w, https://ichef.bbci.co.uk/news/660/cpsdevpb/72c5/test/47abb9d0-5431-11ec-b75b-9f9fe566903d.png 660w"
                    width="999"
                  />
                </picture>
              </div>
              <div
                class="emotion-26 emotion-27"
              />
            </div>
          </div>
          <div
            class="emotion-28 emotion-29"
            dir="ltr"
          >
            <div
              class="emotion-30 emotion-31"
            >
              <a
                aria-labelledby="promo-news-1"
                class="focusIndicatorDisplayBlock emotion-32 emotion-33"
                href="https://www.bbc.co.uk/news"
              >
                <span
                  id="promo-news-1"
                >
                  News homepage
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
